src/rpc/blockchain.cpp

```diff
@@ -1618,7 +1618,7 @@ static RPCHelpMan getchaintxstats()
         }
     }
 
-    const CBlockIndex* pindexPast = pindex->GetAncestor(pindex->nHeight - blockcount);
+    const CBlockIndex* pindexPast = CHECK_NONFATAL(pindex->GetAncestor(pindex->nHeight - blockcount));
```

Comment:
```suggestion
    const CBlockIndex& past_block{*CHECK_NONFATAL(pindex->GetAncestor(pindex->nHeight - blockcount))};
```

nit: no need to keep the pointer while touching this line